ci-operator/step-registry/openshift/e2e/openstack/perfscale/conf-securitygroup/provision/openshift-e2e-openstack-perfscale-conf-securitygroup-provision-commands.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# For disconnected or otherwise unreachable environments, we want to
+# have steps use an HTTP(S) proxy to reach the API server. This proxy
+# configuration file should export HTTP_PROXY, HTTPS_PROXY, and NO_PROXY
+# environment variables, as well as their lowercase equivalents (note
+# that libcurl doesn't recognize the uppercase variables).
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+	# shellcheck disable=SC1090
+	source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+# Getting all worker nodes' names
+mapfile -t WORKER_NODES < <(  oc get nodes -o 'jsonpath={range .items[*]}{.metadata.name}{"\n"}{end}' | grep worker )
+if [ ${#WORKER_NODES[@]} -lt 3 ]
+then 
+	echo "Not enough worker nodes, at least three are needed, can not continue."
+	exit 1
+fi
+
+# Creating the network-perf security group
+NETWORK_PERF_SG=${NETWORK_PERF_SG:-"network-perf-sg"}
+echo "Creating the Security Group $NETWORK_PERF_SG"
+openstack security group create $NETWORK_PERF_SG --description "Security group for running network-perf test on the Worker Nodes"
+openstack security group rule create $NETWORK_PERF_SG --protocol tcp --dst-port 12865:12865 --remote-ip 0.0.0.0/0
+
+# Adding the network-perf security group to the worker nodes
+for i in "${WORKER_NODES[@]}"; do openstack server add security group $i $NETWORK_PERF_SG; done
```

Comment:
Instead of manually adding security groups to the servers, use [the `additionalSecurityGroupIDs` property of machine-pools](https://github.com/openshift/installer/blob/master/docs/user/openstack/customization.md#machine-pools) in `install-config.yaml`.

The way I would go about it is by adding a block [in the `openstack-conf-installconfig` step](https://github.com/openshift/release/blob/7d4ab5483327b1e280528d4ecf1fd46f4464604b/ci-operator/step-registry/openstack/conf/installconfig/openstack-conf-installconfig-commands.sh#L198) that looks a bit like this:

```bash
# in openstack-conf-installconfig-commands.sh

if [[ -d "${SHARED_DIR}/securitygroups" ]]; then
	for sg_id in "${SHARED_DIR}"/securitygroups/*; do
		yq --yaml-output --in-place ".
			| .controlPlane.platform.openstack.additionalSecurityGroupIDs += [ \"$(<"$sg_id")\" ]
			| .compute[0].platform.openstack.additionalSecurityGroupIDs += [ \"$(<"$sg_id")\" ]
		" "$INSTALL_CONFIG"
	done
fi
```

Instead, here you would just create the security group and save its ID into a file:

```bash
sg_id="$(openstack security group create netperf -f value -c id)"
openstack security group rule create "$sg_id" --protocol tcp --dst-port 12865:12865 --remote-ip 0.0.0.0/0

printf '%s' "$sg_id" > "${SHARED_DIR}/securitygroups/netperf"
```